src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -216,59 +241,80 @@ public String toJSON(MaxwellOutputConfig outputConfig) throws IOException {
 
 		g.writeStartObject(); // start of row {
 
+
 		g.writeStringField("database", this.database);
 		g.writeStringField("table", this.table);
 		g.writeStringField("type", this.rowType);
 		g.writeNumberField("ts", this.timestamp);
 
-		if ( outputConfig.includesCommitInfo ) {
-			if ( this.xid != null )
+		if (outputConfig.includesCommitInfo) {
+			if (this.xid != null)
 				g.writeNumberField("xid", this.xid);
 
-			if ( this.txCommit )
+			if (this.txCommit)
 				g.writeBooleanField("commit", true);
 		}
 
-		if ( outputConfig.includesBinlogPosition )
+		if (outputConfig.includesBinlogPosition)
 			g.writeStringField("position", this.nextPosition.getFile() + ":" + this.nextPosition.getOffset());
 
-		if ( outputConfig.includesGtidPosition)
+		if (outputConfig.includesGtidPosition)
 			g.writeStringField("gtid", this.nextPosition.getGtid());
 
-		if ( outputConfig.includesServerId && this.serverId != null ) {
+		if (outputConfig.includesServerId && this.serverId != null) {
 			g.writeNumberField("server_id", this.serverId);
 		}
 
-		if ( outputConfig.includesThreadId && this.threadId != null ) {
+		if (outputConfig.includesThreadId && this.threadId != null) {
 			g.writeNumberField("thread_id", this.threadId);
 		}
 
-		if ( outputConfig.excludeColumns.size() > 0 ) {
+		if (outputConfig.excludeColumns.size() > 0) {
 			// NOTE: to avoid concurrent modification.
 			Set<String> keys = new HashSet<>();
 			keys.addAll(this.data.keySet());
 			keys.addAll(this.oldData.keySet());
 
-			for ( Pattern p : outputConfig.excludeColumns ) {
-				for ( String key : keys ) {
-					if ( p.matcher(key).matches() ) {
+			for (Pattern p : outputConfig.excludeColumns) {
+				for (String key : keys) {
+					if (p.matcher(key).matches()) {
 						this.data.remove(key);
 						this.oldData.remove(key);
 					}
 				}
 			}
 		}
 
-		writeMapToJSON("data", this.data, outputConfig.includesNulls);
+		if(outputConfig.encryptData && !outputConfig.encryptAll){
+			writeEncryptedMapToJSON("data", this.data, outputConfig.includesNulls, outputConfig.encryption_key, outputConfig.secret_key);
+			if(!this.oldData.isEmpty()){
+				writeEncryptedMapToJSON("old", this.oldData, outputConfig.includesNulls, outputConfig.encryption_key, outputConfig.secret_key);
+			}
+		}
+		else {
+			writeMapToJSON("data", this.data, outputConfig.includesNulls);
 
-		if ( !this.oldData.isEmpty() ) {
-			writeMapToJSON("old", this.oldData, true);
+			if (!this.oldData.isEmpty()) {
+				writeMapToJSON("old", this.oldData, true);
+			}
 		}
 
 		g.writeEndObject(); // end of row
 		g.flush();
 
-		return jsonFromStream();
+		if(outputConfig.encryptAll) {
+			return jsonFromStream(outputConfig);
+		}
+		else{
+			return jsonFromStream();
+		}
+	}
+
+	private String jsonFromStream(MaxwellOutputConfig outputConfig){
```

Comment:
this shouldn't be a polymorphic function, should have a different name since it does something fundamentally different